Sources/DatadogObjc/DatadogConfiguration+objc.swift

```diff
@@ -277,6 +277,11 @@ public class DDConfigurationBuilder: NSObject {
         _ = sdkBuilder.set(uploadFrequency: uploadFrequency.swiftType)
     }
 
+    @objc
+    public func set(additionalConfiguration: [String: Any]) {
+        _ = sdkBuilder.set(additionalConfiguration: castAttributesToSwift(additionalConfiguration))
```

Comment:
We shouldn't and we can't use `castAttributesToSwift()` in this case. The `castAttributesToSwift()` wraps `Any` value into `AnyEncodable` wrapper - so it changes its identity. That said, if we pass `"_dd.foo": "bar"` from Obj-c, the downstream assumption in Swift SDK will fail:
```swift
// This will fail, as `additionalConfigurations["_dd.foo"]` is `AnyEncodable`, not `String`:
let bar = additionalConfigurations["_dd.foo"] as? String
```

Instead, we should just pass the `additionalConfiguration` as it is:
```swift
_ = sdkBuilder.set(additionalConfiguration: additionalConfiguration)
```
so the above assumption can pass.

The `castAttributesToSwift()` is only used for capturing serializable values (log / span / rum attributes). In their case, we don't make the `as?` assumption on their identity - we just encode them using encoding implementation from `AnyEncodable`.